File: Team1/RiyaSingh/reverseVowels.java
class Solution {
    static boolean isVowel(char c) { 
        return (c == 'a' || c == 'A' || c == 'e' || c == 'E' || c == 'i' || c == 'I' || c == 'o' || c == 'O' || c == 'u' || c == 'U'); 
    } 
    public String reverseVowels(String s) {
        int left = 0;
        int right = s.length() - 1;
        char[] str = s.toCharArray();
        while(left<right){
            if(!isVowel(s.charAt(left))){
                left++;
                continue;
            }
            else if(!isVowel(s.charAt(right))){
                right--;
                continue;
            }
            char sub = str[left];
            str[left] = str[right];
            str[right] = sub;
            
            left++;
            right--;
        }
        String string = String.copyValueOf(str); 
        return string;
    }
}

File: Team5/ReverseVowels.cpp
class Solution {
private:
  bool m_isVowel(char ch)
  {
    return (
        ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ||
        ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U');
  }

public:
  string reverseVowels(string s)
  {
    int length = s.length();
    int front = 0, end = length - 1;
    char tmp;

    while (front < end)
    {
      if (!m_isVowel(s[front]))
      {
        front++;
      }
      else if (!m_isVowel(s[end]))
      {
        end--;
      }
      else
      {
        tmp = s[front];
        s[front++] = s[end];
        s[end--] = tmp;
      }
    }
    return s;
  }
};
